graphs_scheduling/src/main.jl

```diff
@@ -0,0 +1,132 @@
+include("../../utilities/functions.jl")
+include("../../utilities/auxiliary_functions.jl")
+include("../../utilities/visualization_functions.jl")
+
+# if isdefined(Main, :Dagger) # The Dagger scheduler is already running (?)
+#     ctx = Dagger.Sch.eager_context()
+#     addprocs!(ctx, new_procs)
+# end
+
+# Defining constants
+graph1_path = "./data/sequencer_demo/df_sequencer_demo.graphml"
+graph2_path = "./data/sequencer_demo/another_test_graph.graphml"
+
+LOGS_FILE = timestamp_string("./graphs_scheduling/results/logs/out") * ".dot"
+GRAPH_IMAGE_PATH = timestamp_string("./graphs_scheduling/results/scheduler_images/DAG") * ".png"
+
+OUTPUT_GRAPH_PATH = "./graphs_scheduling/results/parsed_graphs/"
+OUTPUT_GRAPH_IMAGE_PATH = "./graphs_scheduling/results/parsed_graphs_images/"
+
+MAX_GRAPHS_RUN = 3
+
+function execution(graphs_map::Dict{String, String})
+    notifications = Channel{String}(32)
+    dags = Dict{String, TrackedTaskDAG}()
+    running_dags = Set{String}()
+    done_dags = Set{String}()
+
+    schedule_graphs(notifications, graphs_map, dags, running_dags, done_dags)
+    wait_dags_to_finish(notifications, dags, running_dags, done_dags)
+end
+
+function parse_graph(graph_name::String, graph_path::String, output_graph_path::String, output_graph_image_path::String)
+    parsed_graph_dot = timestamp_string("$output_graph_path$graph_name") * ".dot"
+    parsed_graph_image = timestamp_string("$output_graph_image_path$graph_name") * ".png"
+    G = parse_graphml([graph_path])
+    
+    open(parsed_graph_dot, "w") do f
+        MetaGraphs.savedot(f, G)
+    end
+    dot_to_png(parsed_graph_dot, parsed_graph_image)
+    return G
+end
+
+function schedule_graphs(notifications::Channel{String}, graphs_map::Dict{String, String},
+    dags::Dict{String, TrackedTaskDAG}, running_dags::Set{String}, done_dags::Set{String})
+
+    for (g_name, g_path) in graphs_map
+        g = parse_graph(g_name, g_path, OUTPUT_GRAPH_PATH, OUTPUT_GRAPH_IMAGE_PATH)
+        tracked_task_dag = TrackedTaskDAG(g_name, g)
+        dags[get_uuid(tracked_task_dag)] = tracked_task_dag
+
+        while length(running_dags) >= MAX_GRAPHS_RUN
+            wait_dags_to_finish(notifications, dags, running_dags, done_dags, length(running_dags) - MAX_GRAPHS_RUN + 1)
+        end
+        schedule_DAG(tracked_task_dag, notifications, running_dags)
+    end    
+end
+
+function schedule_DAG(tracked_task_dag::TrackedTaskDAG, notifications::Channel{String}, running_dags::Set{String})
+    uuid = get_uuid(tracked_task_dag)
+    start_DAG(tracked_task_dag)
+    push!(running_dags, uuid)
+    println("Dispatcher: graph scheduled - $uuid: $(get_name(tracked_task_dag))")
+    Threads.@spawn begin
+        wait(tracked_task_dag)
+        put!(notifications, string(get_uuid(tracked_task_dag)))
+    end
+end
+
+function wait_dags_to_finish(notifications::Channel{String}, dags::Dict{String, TrackedTaskDAG}, running_dags::Set{String},
+     done_dags::Set{String}, num=length(waiting_notifiers))
```

Comment:
`waiting_notifiers` is not defined

Reply:
Fixed, that was after the latest "cleanup"